### ISSUE TITLE
Failed to start the server daemon after recent update

### ISSUE THREAD

**Issue by n3rada** [BishopFox/sliver#1333]
-🖥️`Linux version 6.1.0-kali9-amd64 (devel@kali.org) (gcc-12 (Debian 12.2.0-14) 12.2.0, GNU ld (GNU Binutils for Debian) 2.40) #1 SMP PREEMPT_DYNAMIC Debian 6.1.27-1kali1 (2023-05-12)`

After updating my sliver version to the v1.5.40, the daemon won't start anymore.

```shell
┌──(kali㉿kali)-[~]
└─$ sliver version  
v1.5.40 - c17c37857e54f0fa202c01cbd00a99a85f5f9f49
```

- Executing with `sudo systemctl start sliver` and then showing the status:
```
● sliver.service - Sliver
     Loaded: loaded (/etc/systemd/system/sliver.service; enabled; preset: disabled)
     Active: activating (auto-restart) (Result: exit-code) since Sun 2023-07-09 18:16:48 CEST; 1s ago
    Process: 11614 ExecStart=/root/sliver-server daemon (code=exited, status=99)
   Main PID: 11614 (code=exited, status=99)
        CPU: 120ms
```

- Executing manually `sudo /root/sliver-server daemon` give me this:
```
goroutine 1 [running]:
runtime/debug.Stack()
        runtime/debug/stack.go:24 +0x65
github.com/bishopfox/sliver/server/cli.glob..func5.1()
        github.com/bishopfox/sliver/server/cli/daemon.go:46 +0x72
panic({0x17d41e0, 0xc0000b8550})
        runtime/panic.go:884 +0x213
github.com/bishopfox/sliver/server/cryptography.ECCServerKeyPair()
        github.com/bishopfox/sliver/server/cryptography/cryptography.go:303 +0xe5
github.com/bishopfox/sliver/server/cli.glob..func5(0xa424fc0?, {0x1a244c9?, 0x0?, 0x0?})
        github.com/bishopfox/sliver/server/cli/daemon.go:54 +0x298
github.com/spf13/cobra.(*Command).execute(0xa424fc0, {0xa50b620, 0x0, 0x0})
        github.com/spf13/cobra@v1.7.0/command.go:944 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0xa424ce0)
        github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.7.0/command.go:992
github.com/bishopfox/sliver/server/cli.Execute()
        github.com/bishopfox/sliver/server/cli/cli.go:145 +0x25
main.main()
        github.com/bishopfox/sliver/server/main.go:43 +0x17
```

Thus, I decided to [compile from source](https://github.com/BishopFox/sliver/wiki/Compile-From-Source) to the previous one make it work again:
![image](https://github.com/BishopFox/sliver/assets/72791564/71aa97a1-23a4-495b-b498-6bc5bf914026)

N.B. It would be helpful to include the latest installation steps for Linux inside the makefile, similar to the steps provided in the install.sh script. This would streamline the installation process for users instead of requiring them to figure it out manually.

**Comment by stephenbradshaw** [comment]
I had the same issue. The release info for [1.5.40](https://github.com/BishopFox/sliver/releases/tag/v1.5.40) mentions backwards incompatible changes, which is why I assume this is not working. I had to use a new sliver configuration directory for versions 1.5.40 and 1.5.39/versions compiled from source to get the later versioned server to work.

**Comment by n3rada** [comment]
Thanks for your answer @stephenbradshaw

What do you mean by using a new configuration directory? And what is the impact from one version to the next?

Could you tell me how you managed the version upgrade, with some example commands? ✍️

**Comment by stephenbradshaw** [comment]
Hi @n3rada 

Essentially I renamed the config directories for the sliver client and server (`~/.sliver` and `~/.sliver-client` by default, although I have the server in a custom location via setting the`SLIVER_ROOT_DIR` environment variable) and let sliver 1.5.40 server and client recreate the application config on first run.

The sliver application needed to be reconfigured from scratch. Its not really an upgrade, but essentially a whole new install. I needed to setup new multiplayer users, generate new payloads, etc. But it "runs". 

Given the release info mentions "backward incompatible changes" I assumed this was intended behaviour. Maybe Im wrong, at the time I was playing with this I was just trying to confirm some behaviour differences between versions on my test instance, and just wanted to get it running ASAP.

Perhaps one of the devs can confirm if an in place upgrade should be possible or not.
